### PR TITLE
{jam,ftjam}: set target-specific macros correctly when cross compiling

### PR DESCRIPTION
**Pull request by impl**
###### Description of changes

Jam embeds some information about the target platform, namely the OS and architecture, directly into the binary. We need to re-evaluate the macros for the target platform when Jam is being used to cross compile. I can successfully cross-compile `p4` with this change, so I think this is hopefully the last change required here.

I'm not super pleased about having to bring `pkgsBuildTarget` into the mix, so if anyone has any better ideas, I'm all ears... :)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/build-managers/jam/default.nix [hunk 1]
```diff
@@ -1,7 +1,9 @@
-{ lib, stdenv, fetchurl, bison, buildPackages }:
+{ lib, stdenv, fetchurl, bison, buildPackages, pkgsBuildTarget }:
```
Comment (AndersonTorres) [inline pkgs/development/tools/build-managers/jam/default.nix:1]: Where is the patch on `all-packages.nix`?

Comment (impl) [reply pkgs/development/tools/build-managers/jam/default.nix:1]: Hmm, I'm not sure I understand what you're asking for -- should I have to set `pkgsBuildTarget` to something explicitly in the `callPackage` for this package in `all-packages.nix`?

`pkgsBuildTarget` seems to work as-is for me, i.e., when I include `jam` in a `nativeBuildInputs` for another package and then cross-compile that package, it correctly represents the set of packages that run on the builder and produce binaries for the cross-compilation target. Does that not work for you? :< If it helps, I was building #190105 as e.g. `pkgsCross.s390x.p4`.

Comment (AndersonTorres) [reply pkgs/development/tools/build-managers/jam/default.nix:1]: Sorry, I was confused. I believed the `ftjam = callPackage ...` line was absent.

**Comment by AndersonTorres** [comment]
It is so wonderful the way we support open source abandonware

**Pull request merged**